### PR TITLE
catch file stream error

### PR DESCRIPTION
**Pull request by sabakugaara**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -103,6 +103,7 @@ module.exports = function (request, options) {
     file.filename = filename
     file.transferEncoding = file.encoding = encoding
     file.mimeType = file.mime = mimetype
+    file.on('error', onEnd);
```
Comment (fengmk2) [inline index.js]: The `error` should handle by the file user, not the busboy inside.

**Comment by fengmk2** [comment]
@sabakugaara I see your issue #28 now.

Can you add a test for this change?

File: index.js (hunk 1 above)
Comment (fengmk2) [inline index.js]: I think this should be `onError`

**Comment by sabakugaara** [comment]
I can't use fake pass-through stream reappear it, only can be reappear by real request. but i can use `part.on('error', function(err) {})` catch it now.  I think it not a bug

**Comment by sabakugaara** [comment]
thanks a lot!

File: index.js (hunk 1 above)
Comment (sabakugaara) [inline index.js]: when I use `part.on('error', function() {})`, how can I throw the error in koa?

**Comment by kevinprotoss** [comment]
has this fix been published in the new version? I think it should be a bug here. If stream error occurs before user handling file event, it will always throw an exception.

**Comment by kevinprotoss** [comment]
However, I think the event handler should be removed if user handle it by himself. 

Currently I have a case. I start to parse multipart request and save returned parts in the koa context. I will not pipe each part immediately. Before I try to use these stream in my middleware, the exception has already be thrown. I have no other chance to catch it.

**Comment by coderhaoxin** [comment]
@kevinprotoss can you add some test cases for this?

**Comment by kevinprotoss** [comment]
I saw this error:

```
events.js:142
      throw er; // Unhandled 'error' event
      ^

Error: Part terminated early due to unexpected end of multipart data
    at /Invend/int_modules/body-parser/node_modules/dicer/lib/Dicer.js:65:36
    at nextTickCallbackWith0Args (node.js:433:9)
    at process._tickCallback (node.js:362:13)
```

I'll try to add a test case.

**Comment by kevinprotoss** [comment]
I found it's hard to reproduce this error in the test case, as @sabakugaara said.

I do the following thing:

In the first middleware
```
let parse = require('co-busboy');
// Initialize co-busboy
let body = {
  parts: parse(this)
};
// Save in the context of Koa.js
this.reqBody = body;
```

In a second middleware
```
console.log('real parsing');
let body = this.reqBody;
let parts = body.parts;
while (!parts.done()) {
  let part = yield parts;
}
```
The error occurs before the second middleware is called.
I guess the stream has been started to pipe before next middleware is called, no matter if I use this stream immediately. If some errors are emitted during this time period, the error event will not be handled.

**Comment by kevinprotoss** [comment]
@coderhaoxin I created a simple gist for reproduce this case. Please have a check.

https://gist.github.com/kevinprotoss/418985fb7ecd7903569b

**Comment by coderhaoxin** [comment]
@kevinprotoss I'm outing for 4 days. Will take a look ASAP. :ok_hand:

**Comment by kevinprotoss** [comment]
Thanks

**Comment by kevinprotoss** [comment]
How about this issue now?

**Comment by coderhaoxin** [comment]
Oh, sorry, my fault. Will take a look these days.

**Comment by coderhaoxin** [comment]
@kevinprotoss Is https://github.com/cojs/busboy/pull/31 helpful?

**Comment by kevinprotoss** [comment]
Thx, it looks helpful. I'll have a further test on my side.

**Comment by kevinprotoss** [comment]
@coderhaoxin I tested on my side. It works now. Thanks in deed again.

**Pull request closed without merging**